### PR TITLE
Grid system

### LINKED ISSUE #318: Grid snapping system
We will want a grid system for snapping.

Longer-term, here are some further features that will encompass the full grid system:

> It should also be displayed over the canvas in user-configurable ways.
> 
> We'll want a concept of document pixels for precise pixel alignment and snapping, drawing pixel art, and previewing a zoomed-in document at its export resolution. (Even though this concept exists for measurements, it does not restrict the rendering of the document at any arbitrarily high resolution, because of Graphite's concept of resolution agnosticism.)
> 
> We will want both pixel and real-world (metric/US) units of measure at arbitrarily large or small scales (millimeters to miles, or even way smaller and larger): #586 
> 
> This suite of grid-related features will live under the Grid checkbox's popover menu. Disabling the checkbox will turn off all rendering of grids. The popover dropdown next to the checkbox allows selectively enabling and disabling specific grid rendering features and changing certain settings like lines or dots for the grid style, the resolution of the grid (based on a chosen unit of measurement, either pixels or real-world units), and even setting the grid to work based on polar coordinates or logarithmic scales in the future.

For the MVP grid snapping, we want:

- [x] Hook up the grid button in the options bar of the document to enable and disable all grid features
- [x] Transforming a layer or point, such as dragging, grabbing (the <kbd>G</kbd> of G/R/S), nudging, etc., should snap (round) it to the nearest integer in document coordinates

### PR DESCRIPTION
**Pull request by OllieDolan**
Tracking Issue: Grid system 
https://github.com/GraphiteEditor/Graphite/issues/318
Grid button triggers snapping points to closest point. 
Closes #318

### CODE REVIEW

File: editor/src/messages/tool/tool_messages/select_tool.rs [hunk 1]
```diff
@@ -626,14 +626,69 @@ impl Fsm for SelectToolFsmState {
...
+							}
+							if !y_aligned && transform_vec_round.x == 0.0 {
+								if transform_vec.y > 0.0 {
+									new_transform_vec.y = new_transform_vec.y.abs() * (1.0 - adjustment.y);
+								} else if transform_vec.y < 0.0 {
+									new_transform_vec.y = new_transform_vec.y.abs() * adjustment.y * -1.0;
+								}
+							}
+
+							responses.add_front(GraphOperationMessage::TransformChange {
```
Comment (OllieDolan) [inline editor/src/messages/tool/tool_messages/select_tool.rs:678]: this responses.add block is repeated twice, could be cleaner to just change "transform_vec" and have the responses.add outside the if and else

File: editor/src/messages/tool/transform_layer/transform_layer_message_handler.rs [hunk 2]
```diff
@@ -83,13 +84,16 @@ impl<'a> MessageHandler<TransformLayerMessage, TransformData<'a>> for TransformL
 						};
 						let points = shape_editor.selected_points();
 
-						*selected.pivot = points.filter_map(get_location).map(count_point).sum::<DVec2>() / point_count as f64;
+						let viewport_pivot = points.filter_map(get_location).map(count_point).sum::<DVec2>() / point_count as f64;
+						*selected.pivot = transform.inverse().transform_point2(viewport_pivot);
+						// *selected.pivot = points.filter_map(get_location).map(count_point).sum::<DVec2>() / point_count as f64;
```
Comment (OllieDolan) [inline editor/src/messages/tool/transform_layer/transform_layer_message_handler.rs:89]: delete comments

**Review by OllieDolan — COMMENTED** [review]
review done

File: editor/src/messages/tool/tool_messages/select_tool.rs (hunk 1 above)
Comment (ChrisMend19) [reply editor/src/messages/tool/tool_messages/select_tool.rs:678]: good idea. Just made the changes

File: .gitattributes [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Requires Git to check out files with the LF (not CRLF) line endings for files it automatically recognizes as being text-based
+# The `*` targets all files
+# The `text=auto` makes it apply a conversion only to files detected as text-based
+# The `eol=lf` sets the conversion to an LF line ending
+# https://git-scm.com/docs/gitattributes
```
Comment (Astropppqv) [inline .gitattributes:5]: - [ ] - The new _@which. @_

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<!-- Please reference any relevant issue number below, optionally with a "Closes"/"Resolves"/"Fixes" prefix -->
```
Comment (Astropppqv) [inline .github/pull_request_template.md:1]: - [ ] I'm **uhci**

File: .vscode/launch.json [hunk 5]
```diff
@@ -7,20 +7,23 @@
 		{
 			"type": "lldb",
 			"request": "launch",
-			"name": "Debug executable 'graphite'",
```
Comment (Astropppqv) [inline .vscode/launch.json:10]: _hik59.   It ___

**Review by Astropppqv — CHANGES_REQUESTED** [review]
601 69

**Pull request closed without merging**